pallets/parachain-staking/src/auto_compounding.rs

```diff
@@ -0,0 +1,296 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Auto-compounding functionality for staking rewards
+
+use crate::pallet::{
+	AutoCompoundingDelegations, BalanceOf, CandidateInfo, Config, DelegatorState, Error, Event,
+	Pallet, Total,
+};
+use crate::types::{Bond, BondAdjust, Delegator};
+use frame_support::ensure;
+use frame_support::traits::Get;
+use frame_support::{dispatch::DispatchResultWithPostInfo, RuntimeDebug};
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::traits::Saturating;
+use sp_runtime::Percent;
+use sp_std::vec::Vec;
+
+/// Represents the auto-compounding amount for a delegation.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, PartialOrd, Ord)]
+pub struct DelegationAutoCompoundConfig<AccountId> {
+	pub delegator: AccountId,
+	pub value: Percent,
+}
+
+/// Sets the auto-compounding value for a delegation.
+pub fn set_delegation_config<AccountId: Eq>(
+	delegations_config: &mut Vec<DelegationAutoCompoundConfig<AccountId>>,
+	delegator: AccountId,
+	value: Percent,
+) {
+	let maybe_delegation = delegations_config
```

Comment:
It's better to use `binary_search_by` at insertion/deletion to the Vector. Like described here: https://substrate.recipes/vec-set.html

I strongly recommend to do this now, because if you want to do it later it will involve a very expensive migration.

Reply:
@nbaztec  Since we want to merge this PR today, if you want I can make this change very quickly